### DISCUSSION TITLE
Example

### DISCUSSION THREAD

**Discussion by masterela** [marcusolsson/grafana-gantt-panel#14] [Q&A] [answered]
Could you please add an example query in your README ? Thanks

**Comment by marcusolsson** [comment, marked as answer]
The query depends on what data source you're using. Is there any data source in particular that you'd like to use this with?

**Reply by masterela** [reply]
Sorry for not posting this in the discussions in the first place. 
I have a postgres DB, and my table has a time column (timestamp with timezone) and a text column. I'm just getting the min and the max time grouped by my text. Tried the time as integer (milliseconds), iso format with and without timezone, but nothing worked.

**Reply by marcusolsson** [reply]
No worries. I only enabled it just a while ago to try it out. Mostly for Q&As that may benefit others (closed issues tend to be harder to find).

There are two things I can think of:

1. You need to set **Format as** to Table
2. The plugin defaults to the first Time field in the query result. Unfortunately, the older data sources, like MySQL and PostgreSQL don't yet fully support Time fields. In fact, the only way I've found to return a typed time field with them is, to name the field "Time". If the response doesn't contain any time fields, you can still set the Start and End dimensions to string or number fields explicitly and they will be converted.

Let me know if that works!

**Reply by ymmt07039** [reply]
my example
![2021-02-12 105517](https://user-images.githubusercontent.com/49381624/107723593-e305f200-6d24-11eb-800e-bf2d839c45ff.png)

**Reply by marcusolsson** [reply]
Looks great!

**Reply by masterela** [reply]
It's telling me "Found no column named time"
This is my query : 

`SELECT phase, to_char (min(time)::timestamp at time zone 'UTC', 'YYYY-MM-DD"T"HH24:MI:SS"Z"') as start_time, to_char (max(time)::timestamp at time zone 'UTC', 'YYYY-MM-DD"T"HH24:MI:SS"Z"') as end_time FROM n_well_1 GROUP BY phase ORDER BY max(time);`

and this is the output of my query in command line:
![image](https://user-images.githubusercontent.com/542567/107869676-0312ee00-6eaa-11eb-93d1-fcc1719524ba.png)

Error screenshot:
![image](https://user-images.githubusercontent.com/542567/107869635-aadbec00-6ea9-11eb-9d8f-ef52f38c069b.png)

Thanks for your help.

**Reply by marcusolsson** [reply]
I'm not SQL ninja, but it seems to me that you have an error in your SQL query at `max(time)`. There's indeed no column called `time`, only `start_time` and `end_time`.

**Reply by ymmt07039** [reply]
Please choose query format  `Table` instead of `Time series`.
Time series needs special format.

**Reply by masterela** [reply]
The query is ok, it works well in postgres and I can see results coming in, I will try the table thing. I think if you select Time series, Grafana will expect a "time" column in the response. So I'll check with Table and let you know. Appreciate the help guys.

**Reply by masterela** [reply]
Here is the response for the query. Changing to Table instead of Time Series removed the "found no time column" thing but still nothing.
This is the response I'm getting from the DB, can you spot something wrong ? 
![image](https://user-images.githubusercontent.com/542567/107922352-514aee80-6f89-11eb-8b7f-91c778512335.png)

This is the query if you're interested:
`SELECT to_char(min(time)::timestamp at time zone 'UTC', 'YYYY-MM-DD" "HH24:MI:SS""') as start_time, to_char (max(time)::timestamp at time zone 'UTC', 'YYYY-MM-DD" "HH24:MI:SS""') as end_time, phase
FROM $well_name
GROUP BY phase
ORDER BY max(time);`

**Reply by marcusolsson** [reply]
The response looks ok to me. Have you tried setting the Dimensions explicitly? The start and end time dimension only defaults to time fields (yours are string fields with time).

**Reply by Andi1986** [reply]
correct me if i'm wrong but i think the date format for start_time and end_time is incorrect. it needs to be either unix timestamp or iso 8601, according to readme

**Reply by marcusolsson** [reply]
Yes. You're right 😅 Good eye!

Try to either convert the timestamp to Unix epochs in ms, or a ISO 8601 string.

**Reply by masterela** [reply]
Yep tried both but it didn't work.

**Reply by Andi1986** [reply]
just to be sure, your timerage on top fits the result data right ? cause the data seems to be pretty old ^^

**Reply by masterela** [reply]
Nope, I'm not using the filter in my query, do you thing it might affect the output ?

**Reply by Andi1986** [reply]
the time range filter will always be applied to the start_time field i think (not sure about that). is your graph still empty when setting time range to "last 5 years" ?

**Reply by masterela** [reply]
I have another table panel next to this one with the exact same query (no time filter) and the data is present and remains unchanged no matter what time filter I've set. My idea was to remove the table and replace it by this Gantt chart because it's easier to spot overlapping phases. I tried multiple time formats, as string (iso format and other common), as integer (milliseconds and seconds), float (with milliseconds after the floating dot) but nothing worked! I would be very grateful if you can post the HTTP response you're getting for your query, I'll format my data in that exact same way and see. Thanks again guys.

**Reply by ymmt07039** [reply]
please refer to my examples.
start_time and end_time are timestamp with timezone.
in your case, Dimensions -> Text  would be `phase` .

![localhost_1](https://user-images.githubusercontent.com/49381624/108168826-420ba280-713b-11eb-8e08-ce4782340450.png)
![localhost_2](https://user-images.githubusercontent.com/49381624/108168837-4637c000-713b-11eb-9d60-fb22dcc5de32.png)

**Reply by masterela** [reply]
![image](https://user-images.githubusercontent.com/542567/108601530-7048e680-73b6-11eb-8a95-516be5286975.png)
It's working, the only missing step was setting the dimensions ! :-(
